app/Http/Controllers/ProductsController.php

```diff
@@ -4,11 +4,38 @@
 
 use Illuminate\Http\Request;
 use App\Product;
+use App\Cart;
+use Session;
+
 class ProductsController extends Controller
 {
     public function index()
     {
     	$products = Product::all();
     	return view('shop.index', compact('products'));
     }
+
+    public function addToCart(Request $request, $id)
+    {
+    	$product = Product::find($id);
+    	$oldCart = Session::has('cart') ? Session::get('cart') : null;
+    	$cart = new Cart($oldCart);
```

Comment:
Can you please give a little information about which part is not acceptable at all. and why ? <3